lighthouse-core/gather/gatherers/seo/tap-targets.js

```diff
@@ -0,0 +1,350 @@
+    if (parentText.length - nodeText.length < 5) {
+      // Parent text mostly consists of this node, so the parent
+      // is not a text block container
+      return false;
+    }
+
+    for (const sibling of node.parentElement.childNodes) {
+      if (sibling === node) {
+        continue;
+      }
+      const siblingTextContent = (sibling.textContent || '').trim();
+      if (sibling.nodeType === Node.TEXT_NODE && siblingTextContent.length > 0) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  if (!isInline(node)) {
+    return false;
+  }
+
+  if (hasTextNodeSiblingsFormingTextBlock(node)) {
+    return true;
+  } else if (node.parentElement) {
+    return nodeIsInTextBlock(node.parentElement);
+  } else {
+    return false;
+  }
+}
+
+/**
+ * @param {Element} node
+ * @returns {boolean}
+ */
+/* istanbul ignore next */
+function nodeIsPositionFixedOrAbsolute(node) {
+  const {position} = getComputedStyle(node);
+  if (position === 'fixed' || position === 'absolute') {
+    return true;
+  }
+  if (node.parentElement) {
+    return nodeIsPositionFixedOrAbsolute(node.parentElement);
+  }
+  return false;
+}
+
+/**
+ * @param {string} str
+ * @param {number} maxLength
+ * @returns {string}
+ */
+/* istanbul ignore next */
+function truncate(str, maxLength) {
+  if (str.length <= maxLength) {
+    return str;
+  }
+  return str.slice(0, maxLength - 1) + '…';
+}
+
+/**
+ * @returns {LH.Artifacts.TapTarget[]}
+ */
+/* istanbul ignore next */
+function gatherTapTargets() {
+  /** @type {LH.Artifacts.TapTarget[]} */
+  const targets = [];
+
+  // @ts-ignore - getElementsInDocument put into scope via stringification
+  Array.from(getElementsInDocument(tapTargetsSelector)).forEach(tapTargetNode => {
+    if (nodeHasParentTapTarget(tapTargetNode)) {
+      // This is usually intentional, either the tap targets trigger the same action
+      // or there's a child with a related action (like a delete button for an item)
+      return;
+    }
+    if (nodeIsInTextBlock(tapTargetNode)) {
+      // Links inside text blocks cause a lot of failures, and there's also an exception for them
+      // in the Web Content Accessibility Guidelines https://www.w3.org/TR/WCAG21/#target-size
+      return;
+    }
+    if (nodeIsPositionFixedOrAbsolute(tapTargetNode)) {
+      // Absolutely positioned elements might not be visible if they have a lower z-index
+      // than other tap targets, but if we don't ignore them we can get false failures.
+      //
+      // TODO: rewrite logic to use elementFromPoint to check if an element is visible
+      // (this should also mean we'd no longer need to check ancestor visible scroll area)
+      return;
+    }
+
+    const visibleClientRects = getVisibleClientRects(tapTargetNode);
+    if (visibleClientRects.length === 0) {
+      return;
+    }
+
+    targets.push({
+      clientRects: visibleClientRects,
+      snippet: truncate(tapTargetNode.outerHTML, 700),
```

Comment:
700 sounds a bit large, were the smaller numbers you tried just not enough to figure out where it is?

Reply:
Yeah, sometimes the outerHTML contains a lot of noise (lots of classes, children) and cutting the outerHTML off early risks losing information that's needed to identify the element.

So it's a trade-off between result size and ensuring every element is identifiable. Maybe let's go down to 300?

We should look into [showing something other than plain outerHTML](https://github.com/GoogleChrome/lighthouse/issues/6683) at some point.

Reply:
Changed it to 300 now.